OpenRA.Platforms.Default/ThreadedGraphicsContext.cs

```diff
@@ -389,24 +405,24 @@ public IFrameBuffer CreateFrameBuffer(Size s, Color clearColor)
 			return Send(getCreateFrameBuffer, (s, clearColor));
 		}
 
-		public IShader CreateShader(string name)
+		public IShader CreateShader(IShaderBindings bindings)
 		{
-			return Send(getCreateShader, name);
+			return Send(getCreateShader, bindings);
```

Comment:
Yes, anything that uses `Send` blocks since it has to be able to relay the result to the caller. But those aren't called very often. The methods than can use `Post` and avoid blocking are the ones that get called during a typical frame and where the gains come from.

Reply:
Yes, anything that uses Send blocks since it has to be able to relay the result to the caller. But those aren't called very often. The methods than can use Post and avoid blocking are the ones that get called during a typical frame and where the gains come from.